src/main/java/liuni/TimeCache.java

```diff
@@ -0,0 +1,84 @@
+package liuni;
+
+import liuni.models.TwitterTweetModel;
+
+import java.util.Date;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public class TimeCache {
+    private final int TIME_CONVERSION = 1000;
+    private long timeToLive;
+    private int maxEntries;
+    private Map<String, TimedList> cache;
+
+    protected class TimedList {
+        private Date creationDate;
+        private List<TwitterTweetModel> data;
+        private boolean important;
+
+        public TimedList(Date date, List<TwitterTweetModel> list, boolean imp) {
+            creationDate = date;
+            data = list;
+            important = imp;
+        }
+
+        public boolean outOfDate() {
+            Date now = new Date();
+            long different = now.getTime() - creationDate.getTime();
+            return different > timeToLive;
+        }
+    }
+
+    public TimeCache(long maxTimeSec, int maxItems) {
+        maxEntries = maxItems;
+        timeToLive = maxTimeSec * TIME_CONVERSION;
+        createNewCache();
+    }
+
+    public void createNewCache() {
+        cache = new LinkedHashMap(maxEntries + 1, 0.75F, true) {
+            @Override
+            protected boolean removeEldestEntry(Map.Entry eldest) {
+                removeOldEntries();
+                return size() > maxEntries && !((TimedList) eldest.getValue()).important;
+            }
+        };
+    }
+
+    private void removeOldEntries() {
+        cache.entrySet().removeIf(setEntry -> setEntry.getValue().outOfDate());
+    }
+
+    public List<TwitterTweetModel> getEntry(String key) {
+        removeOldEntries();
+        if (isEntry(key)) {
+            return cache.get(key).data;
+        }
+        return null;
+    }
+
+    public List<TwitterTweetModel> putEntry(String key, List<TwitterTweetModel> value) {
+        if (key.equals("timeline")) {
```

Comment:
why the need to distinguish between timeline?

Reply:
To list the timeline entry as important so that it is not removed if you try to add more entries to the cache

Reply:
is it not the case where the oldest value gets removed anyway?